intelmq/lib/mixins/cache.py

```diff
@@ -51,6 +67,20 @@ def cache_set(self, key: str, value: Any, ttl: Optional[int] = None):
         if self.redis_cache_ttl:
             self.__redis.expire(key, self.redis_cache_ttl)
 
+    def cache_put(self, value: dict) -> int:
+        # Returns the length of the list after pushing
+        size = self.__redis.lpush(self.bot_id, json.dumps(value))
+        return size
+
+    def cache_length(self) -> int:
+        return self.__redis.llen(self.bot_id)
+
+    def cache_pop(self) -> dict:
+        data = self.__redis.rpop(self.bot_id)
+        if data is None:
+            return None
+        return json.loads(data)
```

Comment:
```suggestion
    def cache_rpop(self, key: str) -> dict:
        data = self.__redis.rpop(key)
        return data
```

The new function is missing in the docs in `docs/dev/bot-development.md`.